### PR TITLE
feat: fix categories being set on offbudget transactions

### PR DESCRIPTION
**Pull request by UnderKoen**

### CODE REVIEW

**Comment by youngcw** [comment]
This will be fine for now, but another thing that would be good would be for transactions to get stripped of categories when being added to off budget accounts.  People occasionally get categories is their offbudget accounts and it causes issues.

**Comment by UnderKoen** [comment]
@youngcw 
> This will be fine for now, but another thing that would be good would be for transactions to get stripped of categories when being added to off budget accounts. People occasionally get categories is their offbudget accounts and it causes issues.

Will create seprate issue for that

File: packages/loot-core/src/server/accounts/transactions.ts [hunk 1]
```diff
@@ -96,7 +102,7 @@ export async function batchUpdateTransactions({
     if (updated) {
       await Promise.all(
         updated.map(async t => {
-          if (t.account) {
```
Comment (youngcw) [inline packages/loot-core/src/server/accounts/transactions.ts:99]: So updating a transaction should have already removed the category?  Why was it not working?  or is this just to catch existing things?

Comment (UnderKoen) [reply packages/loot-core/src/server/accounts/transactions.ts:99]: Only if the account was changed, so you could add a category to a transaction after it was moved or created in the off budget account

Comment (UnderKoen) [reply packages/loot-core/src/server/accounts/transactions.ts:99]: Actually I have no clue why it was not working. My change here wasn't working but I also couldn't find an method of adding a category after creation.

**Comment by UnderKoen** [comment]
@youngcw Fixxed it!

File: packages/loot-core/src/server/accounts/transfer.ts [hunk 2]
```diff
@@ -31,7 +31,12 @@ async function clearCategory(transaction, transferAcct) {
   // transferring from an on-budget to off-budget account
```
Comment (youngcw) [inline packages/loot-core/src/server/accounts/transfer.ts:31]: this comment looks wrong

Comment (UnderKoen) [reply packages/loot-core/src/server/accounts/transfer.ts]: This better?

Comment (youngcw) [reply packages/loot-core/src/server/accounts/transfer.ts]: nice

**Comment by youngcw** [comment]
I like how you started trying to fix the reports 😄

**Review by youngcw — APPROVED** [review]

**Pull request merged**